crates/cdk/src/wallet/client.rs

```diff
@@ -367,3 +373,90 @@ impl HttpClient {
         }
     }
 }
+
+/// Http Client Methods
+#[async_trait]
+pub trait HttpClientMethods: Debug {
+    /// Get Active Mint Keys [NUT-01]
+    async fn get_mint_keys(&self, mint_url: MintUrl) -> Result<Vec<KeySet>, Error>;
+
+    /// Get Keyset Keys [NUT-01]
+    async fn get_mint_keyset(&self, mint_url: MintUrl, keyset_id: Id) -> Result<KeySet, Error>;
+
+    /// Get Keysets [NUT-02]
+    async fn get_mint_keysets(&self, mint_url: MintUrl) -> Result<KeysetResponse, Error>;
+
+    /// Mint Quote [NUT-04]
+    async fn post_mint_quote(
+        &self,
+        mint_url: MintUrl,
+        amount: Amount,
+        unit: CurrencyUnit,
+        description: Option<String>,
+    ) -> Result<MintQuoteBolt11Response, Error>;
+
+    /// Mint Quote status
+    async fn get_mint_quote_status(
+        &self,
+        mint_url: MintUrl,
+        quote_id: &str,
+    ) -> Result<MintQuoteBolt11Response, Error>;
+
+    /// Mint Tokens [NUT-04]
+    async fn post_mint(
+        &self,
+        mint_url: MintUrl,
+        quote: &str,
+        premint_secrets: PreMintSecrets,
+    ) -> Result<MintBolt11Response, Error>;
+
+    /// Melt Quote [NUT-05]
+    async fn post_melt_quote(
+        &self,
+        mint_url: MintUrl,
+        unit: CurrencyUnit,
+        request: Bolt11Invoice,
+        mpp_amount: Option<Amount>,
+    ) -> Result<MeltQuoteBolt11Response, Error>;
+
+    /// Melt Quote Status
+    async fn get_melt_quote_status(
+        &self,
+        mint_url: MintUrl,
+        quote_id: &str,
+    ) -> Result<MeltQuoteBolt11Response, Error>;
+
+    /// Melt [NUT-05]
+    /// [Nut-08] Lightning fee return if outputs defined
+    async fn post_melt(
+        &self,
+        mint_url: MintUrl,
+        quote: String,
+        inputs: Vec<Proof>,
+        outputs: Option<Vec<BlindedMessage>>,
+    ) -> Result<MeltQuoteBolt11Response, Error>;
+
+    /// Split Token [NUT-06]
+    async fn post_swap(
+        &self,
+        mint_url: MintUrl,
+        swap_request: SwapRequest,
+    ) -> Result<SwapResponse, Error>;
+
+    /// Get Mint Info [NUT-06]
+    async fn get_mint_info(&self, mint_url: MintUrl) -> Result<MintInfo, Error>;
+
+    /// Spendable check [NUT-07]
+    async fn post_check_state(
+        &self,
+        mint_url: MintUrl,
+        ys: Vec<PublicKey>,
+    ) -> Result<CheckStateResponse, Error>;
+
+    /// Restore request [NUT-13]
+    async fn post_restore(
+        &self,
+        mint_url: MintUrl,
+        request: RestoreRequest,
```

Comment:
Nice I like this PR. 

This question stems from my mistake not a change in this pr and has been in the back of my mind for awhile but think this pr is a good chance to clean it up. 

Some of the methods take the request Struct like `post_swap` but then some of the methods take the request fields and internally construct the request struct like `post_melt`. I think we should unify this and always take the request struct. What do you think does that make sense?

Reply:
Good question. Since this is intended to be a low-level trait that isn't required for most implementations, I prefer passing the struct to a method like `post_swap`. In case any future functions have a lot of fields, it makes the most sense to standardize them to this. But I can be convinced the other way because, in general, I prefer not having to create structs to call a function, but in this case, I think it makes the most sense.

Reply:
Agree think the struct makes most sense here.

Reply:
Updated